### PR TITLE
Log socket protocol types when sweeping ghost connections

### PR DESCRIPTION
**Pull request by Kaiepi**

### CODE REVIEW

**Comment by Zarel** [comment]
I don't think `Monitor.debug` is logged anywhere in PS...

**Comment by Zarel** [comment]
Yeah, Main's loglevel is like 3

**Comment by Zarel** [comment]
Not like the console's output is logged anywhere, either.

**Comment by Kaiepi** [comment]
How should I go about logging this?

**Comment by Kaiepi** [comment]
Updated

File: sockets.js [hunk 1]
```diff
@@ -404,10 +405,14 @@ if (cluster.isMaster) {
 	const channels = new Map();
 	/** @type {Map<string, Map<string, string>>} */
 	const subchannels = new Map();
+	/** @type {WriteStream} */
+	const logger = FS('logs/sockets').createAppendStream();
```
Comment (Zarel) [inline sockets.js]: Is this once per socket? You should probably append a process ID so you don't have multiple processes fighting to append to the same stream.

Comment (Zarel) [reply sockets.js]:
```suggestion
	const logger = FS(`logs/sockets-${process.pid}`).createAppendStream();
```

File: sockets.js [hunk 2]
```diff
@@ -404,10 +405,14 @@ if (cluster.isMaster) {
 	const channels = new Map();
 	/** @type {Map<string, Map<string, string>>} */
 	const subchannels = new Map();
+	/** @type {WriteStream} */
+	const logger = FS(`logs/sockets-${process.pid}`).createAppendStream();
 
 	// Deal with phantom connections.
 	const sweepSocketInterval = setInterval(() => {
 		sockets.forEach(socket => {
+			logger.write(`Found a ghost connection with a protocol of ${socket.protocol}\n`);
```
Comment (Zarel) [inline sockets.js]: Notice how this is for every socket ever, not just for ghost connections...

**Pull request merged**